pkg/bgpv1/gobgp/state.go

```diff
@@ -0,0 +1,158 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package gobgp
+
+import (
+	"context"
+	"time"
+
+	gobgp "github.com/osrg/gobgp/v3/api"
+
+	"github.com/cilium/cilium/api/v1/models"
+	"github.com/cilium/cilium/pkg/bgpv1/agent"
+)
+
+// GetPeerState invokes goBGP ListPeer API to get current peering state.
+func (sc *ServerWithConfig) GetPeerState(ctx context.Context) ([]*models.BgpPeer, error) {
+	var data []*models.BgpPeer
+	fn := func(peer *gobgp.Peer) {
+		if peer == nil {
+			return
+		}
+
+		peerState := &models.BgpPeer{}
+
+		if peer.Conf != nil {
+			peerState.LocalAsn = int64(peer.Conf.LocalAsn)
+			peerState.PeerAddress = peer.Conf.NeighborAddress
+			peerState.PeerAsn = int64(peer.Conf.PeerAsn)
+		}
+
+		if peer.State != nil {
+			peerState.SessionState = toAgentSessionState(peer.State.SessionState).String()
+
+			// Uptime is time since session got established.
+			// It is calculated by difference in time from uptime timestamp till now.
+			// Time is rounded to second precision.
```

Comment:
Why do we round to second precision here? 
My assumption is because we want the CLI output to present seconds?
If that is the case, this makes an assumption of how `GetPeers` will be used, and gives me a slight code smell.

Reply:
Agreed, I will leave it in nano seconds. Just for CLI output it can be rounded to second precision. In json output, it will be actual nano-second precision.

Reply:
Most of the vendors give second level precision for uptime of BGP state. 

I was looking at bgp yang model - precision in the RFC is also upto seconds. 
https://www.ietf.org/archive/id/draft-ietf-idr-bgp-model-16.html 
Value to search : last-established